go/fixchain/hash.go

```diff
@@ -0,0 +1,60 @@
+package fixchain
+
+import (
+	"crypto/sha256"
+	"sort"
+
+	"github.com/google/certificate-transparency/go/x509"
+)
+
+const hashSize = sha256.Size
+
+var newHash = sha256.New
+
+func hash(c *x509.Certificate) (hash [hashSize]byte) {
+	copy(hash[:], newHash().Sum(c.Raw))
+	return
+}
+
+func hashChain(ch []*x509.Certificate) (hash [hashSize]byte) {
+	h := newHash()
+	for _, c := range ch {
+		h.Write(newHash().Sum(c.Raw))
+	}
+	copy(hash[:], h.Sum(nil))
+	return
+}
+
+func hashBag(chain []*x509.Certificate) [hashSize]byte {
```

Comment:
At least comment that it would hash differently for chains with dupes, but not for chains with different orders (which is why it is a bag rather than a set).

Though I question why we would ever want a bag rather than a set!